### ISSUE TITLE
UINavigationController and UITabBarController Categories

### ISSUE THREAD

**Issue by OshitaYoichi** [gdavis/FGallery-iPhone#32]
The FGallery categories for UINavigationController and UITabBarController contain a shouldAutorotateToInterfaceOrientation: method.

This is terribly wrong. It took me hours to debug this.

The issue here is that when you import the gallery module sources and have -all_load linker flag this overrides the default behaviour of the nav bar controller and tab bar controller and the shouldAutorotateToInterfaceOrientation: event of the view controllers doesn't get called, the category one overrides it.

If you want to always rotate the gallery screen you should delegate that responsibility to the developer who uses the library.

Kinds regards,
Ivan.

**Comment by KoCMoHaBTa** [comment]
I can confirm that. The problem occurs even without any linker flags.

**Comment by gdavis** [comment]
Agreed. This is some old code that served as a hack at the time and should now be handled differently or completely removed. Thanks for bringing it up and sorry for the issues
